Src/Chill/GivenWhenThen.cs

```diff
@@ -95,25 +96,25 @@ protected void WhenLater(Func<Task<TResult>> whenFunc)
         /// <param name="deferredExecution">Should the test be executed immediately or be deferred?</param>
         protected void When(Func<Task<TResult>> whenFunc, bool? deferredExecution = null)
         {
-            this.When(() => whenFunc.ExecuteInDefaultSynchronizationContext(), deferredExecution);
+            When(() => whenFunc.ExecuteInDefaultSynchronizationContext(), deferredExecution);
         }
 
         /// <summary>
         /// Records an asynchronous precondition.
         /// </summary>
         public void Given(Func<Task> givenActionAsync)
         {
-            this.Given(() => givenActionAsync.ExecuteInDefaultSynchronizationContext());
+            Given(() => givenActionAsync.ExecuteInDefaultSynchronizationContext());
```

Comment:
We already call ExeuteInDefaultSync... inside TriggerTest in the base class. This can be replaced with only GetAwaiter().GetResult()